hivelink-core/src/main/java/org/apache/iceberg/hivelink/core/HiveMetadataPreservingTableOperations.java

```diff
@@ -195,17 +195,36 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     ReentrantLock tableLevelMutex = commitLockCache.get(fullName, t -> new ReentrantLock(true));
     tableLevelMutex.lock();
     try {
+      LOG.warn(
+          "In thread {}, starting to acquire a lock for table {}",
+          Thread.currentThread(),
+          fullName);
       lockId = Optional.of(acquireLock());
+      LOG.warn(
+          "In thread {}, acquired lock id: {} for table {}",
+          Thread.currentThread(),
+          lockId.get(),
+          fullName);
       // TODO add lock heart beating for cases where default lock timeout is too low.
       Table tbl;
       // [LINKEDIN] Instead of checking if base != null to check for table existence, we query
       // metastore for existence
       // base can be null if not Iceberg metadata exists, but Hive table exists, so we want to get
       // the current table
       // definition and not create a new definition
+      LOG.warn("In thread {}, checking if table exists {}", Thread.currentThread(), fullName);
```

Comment:
Nit: should we change this to level info, it seems like not strongly related to the locking issue

Reply:
I think let's consistency use `warn`, so that we can analyze the issue easier, also when we are done with finding out the RC, we should revert this PR.

Reply:
Ack, make sense